File: backend/src/main/java/com/wintina/blog/entity/Post.java
package com.wintina.blog.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@TableName("post")
public class Post {
    @TableId(type = IdType.AUTO, value = "id")
    private Long id;

    @TableField(value = "user_id")
    private Long userId;

    // 关联用户表
    @TableField(exist = false)
    private User user;

    @TableField(value = "category_id")
    private Long categoryId;

    // 关联分类表
    @TableField(exist = false)
    private Category category;

    @TableField(value = "title")
    private String title;

    @TableField(value = "slug")
    private String slug;

    @TableField(value = "summary")
    private String summary;

    @TableField(value = "content")
    private String content;

    @TableField(value = "content_html")
    private String contentHtml;

    @TableField(value = "cover_image")
    private String coverImage;

    @TableField(value = "status")
    private Integer status = 1;

    @TableField(value = "is_top")
    private Integer isTop = 0;

    @TableField(value = "is_recommend")
    private Integer isRecommend = 0;

    @TableField(value = "allow_comment")
    private Integer allowComment = 1;

    @TableField(value = "view_count")
    private Integer viewCount = 0;

    @TableField(value = "like_count")
    private Integer likeCount = 0;

    @TableField(value = "comment_count")
    private Integer commentCount = 0;

    @TableField(value = "collect_count")
    private Integer collectCount = 0;

    @TableField(value = "sort_order")
    private Integer sortOrder = 0;

    @TableField(value = "seo_keywords")
    private String seoKeywords;

    @TableField(value = "seo_description")
    private String seoDescription;

    @TableField(value = "publish_time")
    private LocalDateTime publishTime;

    @TableField(value = "create_time")
    private LocalDateTime createTime;

    @TableField(value = "update_time")
    private LocalDateTime updateTime;

    @TableLogic
    @TableField(value = "is_deleted")
    private Integer isDeleted = 0;


}

File: web/src/types/post.ts
import type { PageResult } from './common'

export interface PostQueryParams {
  current?: number
  size?: number
  categoryId?: number
  tagId?: number
  keyword?: string
}

export interface PostListItem {
  id: number
  title: string
  slug?: string
  summary?: string
  coverImage?: string
  status?: number
  viewCount?: number
  likeCount?: number
  commentCount?: number
  readTimeMinutes?: number
  userId?: number
  authorName?: string
  authorAvatar?: string
  categoryId?: number
  categoryName?: string
  tagIds?: number[]
  tagNames?: string[]
  publishTime?: string
  createTime?: string
  updateTime?: string
}

export interface TocItem {
  id: string
  text: string
  level: number
}

export interface RelatedPostItem {
  id: number
  title: string
  coverImage?: string
  publishTime?: string
  viewCount?: number
}

export interface PostDetail extends PostListItem {
  content?: string
  contentHtml?: string
  allowComment?: number
  isTop?: number
  isRecommend?: number
  collectCount?: number
  sortOrder?: number
  seoKeywords?: string
  seoDescription?: string
  authorBio?: string
  authorPostCount?: number
  toc?: TocItem[]
  relatedPosts?: RelatedPostItem[]
}

export interface SavePostPayload {
  title: string
  summary?: string
  content: string
  contentHtml?: string
  categoryId: number
  status: number
  slug?: string
  coverImage?: string
  tagIds?: number[]
  allowComment?: number
  isTop?: number
  isRecommend?: number
  sortOrder?: number
  seoKeywords?: string
  seoDescription?: string
}

export interface PostEditorCategory {
  id: number
  name: string
}

export interface PostEditorTag {
  id: number
  name: string
  color?: string
}

export interface PostEditorStatusOption {
  value: number
  label: string
}

export interface PostEditorOptions {
  categories: PostEditorCategory[]
  tags: PostEditorTag[]
  statusOptions: PostEditorStatusOption[]
}

export type PostPageResult = PageResult<PostListItem>
